packages/provider/src/TiptapCollabProvider.ts

```diff
@@ -297,12 +308,22 @@ export class TiptapCollabProvider extends HocuspocusProvider {
 
     // if the first comment of a thread is deleted we also
     // delete the thread itself as the source comment is gone
-    if (commentIndex === 0) {
+    if (commentIndex === 0 && deleteThread) {
       this.deleteThread(threadId)
       return
     }
 
     if (commentIndex > 0) {
+      const comment = thread.get('comments').get(commentIndex)
+      const newComment = new Y.Map()
+
+      newComment.set('id', comment.get('id'))
+      newComment.set('createdAt', comment.get('createdAt'))
+      newComment.set('updatedAt', (new Date()).toISOString())
+      newComment.set('data', comment.get('data'))
+      newComment.set('content', null)
```

Comment:
This defeats the purpose of tombstoning though? You lose access to what the comment had said.

The only way that a user would be able to see the tombstoned comment is that they would call a method that allows them access to the deleted comment. I think this should be left up to the application, because how would they mark comments as deleted but still preserve their history?

Reply:
I understood tombstoning like deleting a specific item of content but keeping the entry to preserve the item itself but not it's content.

Why would I delete something if the content would still be present anyways and I as the content author had no way to get rid of it again?

Reply:
> Why would I delete something if the content would still be present anyways
That's what tomb-stoning is, you are making the tombstone for the data before it is actually dead 😂 

The idea is that marking the content as deleted (by moving it to another field) allows you to still find the data when you need to, imagine an archival snapshot where you want to see what that resolved comment said. While, also allowing it to be effectively removed in normal circumstances (i.e. showing what the current comments are). Which is why the API needs to be opt-in to see deleted content.

Ain't no one looking through yjs data, and even if they did, technically data is never actually deleted in yjs either haha. We can also have what is called a hard deletion to do what you are proposing, which I would again make an opt-in flag for.

Reply:
Okay so my idea would be:

1. Keep the content on the actual yjs document but
2. When queried via `getThreads` or the `getComments` functions, the content will be replaced so it's not available directly to clients

Reply:
That way we could use it later on / restore it but still keep the stuff deleted kind of private.

Reply:
It would look like this to me:

`getAllComments(includeDeleted?: boolean)` where includeDeleted as true gives you the whole deleted comment
`deleteComment({ deleteThread?: boolean; deleteContent?: boolean; })` where `deleteThread` does what it does right now, and `deleteContent` will default to false, but if they actually want a hard delete then they can do that.

The point of doing it like that is that it is opt-in to read deleted content for the reasons I stated above, and opt-in to force remove content. This is the double opt-in I was referring to when we spoke.

Reply:
Any method that is exposed should be by default operating only on the non-deleted content. But, some can expose an opt-in way of operating on the deleted content (like getAllComments in my example)